data-access/src/object_store/mod.rs

```diff
@@ -80,7 +81,7 @@ pub trait ObjectStore: Sync + Send + Debug {
         prefix: &str,
         suffix: &str,
     ) -> Result<FileMetaStream> {
-        let file_stream = self.list_file(prefix).await?;
+        let file_stream = self.glob_file(prefix).await?;
```

Comment:
It seems confusing if `list_file` doesn't resolve globs, but `list_file_with_suffix` does. Perhaps we could add a `glob_file_with_suffix`, to avoid changing the behavior of a pre-existing method? This could then do something like

```
match (is_glob_path(prefix), suffix.is_empty()) {
  (false, true) => self.list_file(prefix).await,
  (false, false) => self.list_file_with_suffix(prefix, suffix).await,
  (true, _) => { // glob logic }
}
```

`glob_file` could then just be

```
async fn glob_file(&self, prefix: &str) -> Result<FileMetaStream> {
  self.glob_file_with_suffix(prefix, "").await
}
```